src/plugins/graph/utilities/graph-utils.ts

```diff
@@ -140,37 +141,203 @@ export interface IMatchCirclesProps {
+  styleOuterCircles(outerCircles, dataset);
+}
+
+function applySelectedClassToCircles(selection: DotSelection, dataset?: IDataSet){
+  selection
+    .classed('selected', (aCaseData: CaseData) => !!dataset?.isCaseSelected(aCaseData.caseID));
+}
+
+function styleOuterCircles(outerCircles: any, dataset?: IDataSet){
+  if (dataset?.selection.size === 0) {
+    outerCircles
+      .attr('r', (aCaseData: CaseData) => {
+        return 0;
+      });
+  }
+  else {
+    outerCircles
+      .attr('r', (aCaseData: CaseData) => {
+        return (dataset?.isCaseSelected(aCaseData.caseID)) ? 10 : 0;
+      })
+      .style('fill', (aCaseData: CaseData) => {
+        return (dataset?.isCaseSelected(aCaseData.caseID)) && selectedOuterCircleFillColor;
+      })
+      .style('stroke', (aCaseData: CaseData) =>{
+        return selectedOuterCircleStrokeColor;
+      })
+      .style('opacity', 0.5);
+  }
+}
+
+
+export interface ISetPointCoordinates {
+  dataset?: IDataSet
+  dotsRef: IDotsRef
+  selectedOnly?: boolean
+  pointRadius: number
+  selectedPointRadius: number
+  pointColor: string
+  pointStrokeColor: string
+  getPointColorAtIndex?: (index: number) => string
+  getScreenX: ((anID: string) => number | null)
+  getScreenY: ((anID: string, plotNum?:number) => number | null)
+  getLegendColor?: ((anID: string) => string)
+  enableAnimation: React.MutableRefObject<boolean>
+}
+
+export function setPointCoordinates(props: ISetPointCoordinates) {
+
+  const setPoints = (radius: number) => {
+    if (theSelection?.size()) {
+      theSelection
+        .attr('cx', (aCaseData: CaseData) => {
+          return getScreenX(aCaseData.caseID);
+        })
+        .attr('cy', (aCaseData: CaseData) => {
+          return getScreenY(aCaseData.caseID, aCaseData.plotNum);
+        })
+        .attr('r', (aCaseData: CaseData) => {
+          return radius;
+        })
+        .style('fill', (aCaseData: CaseData) => {
+          return lookupLegendColor(aCaseData);
+        })
+        .style('stroke', (aCaseData: CaseData) =>{
+          if (getPointColorAtIndex){
+            return getPointColorAtIndex(aCaseData.plotNum);//border color of inner dot should be same color as legend
+          } else {
+            return pointColor;
+          }
+        })
+        .style('stroke-width', (aCaseData: CaseData) =>{
+          return (dataset?.isCaseSelected(aCaseData.caseID))
+                 ? selectedStrokeWidth : defaultStrokeWidth;
+        });
+    }
+  };
+
+  const lookupLegendColor = (aCaseData: CaseData) => {
+    const id = aCaseData.caseID;
+    const isSelected = dataset?.isCaseSelected(id);
+    const legendColor = getLegendColor ? getLegendColor(id) : '';
+    if (getPointColorAtIndex) {
+      if (legendColor !== '') {
+        return legendColor;
+      } else if (isSelected ||(aCaseData.plotNum && getPointColorAtIndex)) {
+        return getPointColorAtIndex(aCaseData.plotNum);
+      } else {
+        return pointColor;
```

Comment:
```suggestion
    if (legendColor !== '') {
      return legendColor;
    } else if (getPointColorAtIndex && aCaseData.plotNum) {
      return getPointColorAtIndex(aCaseData.plotNum);
    } else if (isSelected) {
      return defaultSelectedColor;
    } else {
      return pointColor;
```

Reply:
I was looking through the code yesterday evening and realized you resolved this without taking my suggestion. You should always either accept my suggestions, or explain why you don't think they're a good idea. I'm never going to spend my time giving you a suggestion without having a reason and thinking through the code. In this case, your logic was unnecessarily complex, leading to duplicated code and spurious code that would be confusing to engineers that stumbled upon it in the future.